src/coreclr/vm/codeman.cpp

```diff
@@ -1448,6 +1448,21 @@ void EEJitManager::SetCpuInfo()
     {
         CPUCompileFlags.Set(InstructionSet_X86Serialize);
     }
+
+    if (((cpuFeatures & XArchIntrinsicConstants_Avx10v1) != 0) && CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableAVX10v1))
+    {
+        CPUCompileFlags.Set(InstructionSet_AVX10v1);
+    }
+
+    if (((cpuFeatures & XArchIntrinsicConstants_Avx10v1_V256) != 0) && CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableAVX10v1))
```

Comment:
I don't think we need this `CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableAVX10v1)` check.

The `EnsureValidInstructionSetSupport` call below should ensure it is correct provided the `implication` set in `InstructionSetDesc.txt` are correct

Reply:
you mean here we only check if XArchIntrinsicConstants are set for Avx10v1_V256/512 and make `Avx10v1_V512` implies `Avx10v1_V256` and `Avx10v1_V256` implies `Avx10v1`?

This is confirmed in Avx10 spec doc:
>Developers can assume that the highest supported vector length for a processor implies that all lesser vector
lengths are also supported. Scalar Intel AVX-512 instructions will be supported independent of the maximum
vector width.

Reply:
Correct. Once the implications are correct, then `EnusreValidInstructionSetSupport` should let everything else flow out, so we only need to check for `CLRConfig::EXTERNAL_EnableAVX10v1` the once